datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -973,6 +982,24 @@ fn create_schema(
     Ok(Schema::new(fields))
 }
 
+/// returns schema with dictionary group keys materialized as their value types
+/// The actual convertion happens in `RowConverter` and we don't do unnecessary
+/// conversion back into dictionaries
+fn materialize_dict_group_keys(schema: &Schema, group_count: usize) -> Schema {
```

Comment:
`GroupValues` is a trait and there are two implementations of it -- `GroupValueRows` and `GroupValuesPrimitive`

This conversion only applies to `GroupValueRows` (though `GroupValuesPrimitive` won't be used for input dictionary columns anyways)

I wonder if there is some way to move the conversion into `GroupValueRows` so it is clearer from the context when this is needed or not

Reply:
I'll take a look